atomicfu/src/jvmMain/kotlin/kotlinx/atomicfu/Trace.kt

```diff
@@ -24,41 +24,74 @@ public actual val traceFormatDefault: TraceFormat =
     if (getSystemProperty("kotlinx.atomicfu.trace.thread") != null) TraceFormatThread() else TraceFormat()
 
 private class TraceFormatThread : TraceFormat() {
-    override fun format(index: Int, text: String): String =
+    override fun format(index: Int, text: Any): String =
         "$index: [${Thread.currentThread().name}] $text"
 }
 
 private class NamedTrace(
     private val trace: TraceBase, 
     private val name: String
 ) : TraceBase() {
-    override fun append(text: String) = trace.append("$name.$text")
+    override fun append(arg1: Any) = trace.append("$name.$arg1")
+
+    override fun append(arg1: Any, arg2: Any) = trace.append("$name.$arg1", "$name.$arg2")
+
+    override fun append(arg1: Any, arg2: Any, arg3: Any) =
+            trace.append("$name.$arg1", "$name.$arg2", "$name.$arg3")
+
+    override fun append(arg1: Any, arg2: Any, arg3: Any, arg4: Any) =
+            trace.append("$name.$arg1", "$name.$arg2", "$name.$arg3", "$name.$arg4")
+
     override fun toString(): String = trace.toString()
 }
 
 private class TraceImpl(size: Int, val format: TraceFormat) : TraceBase() {
     init { require(size >= 1) }
     private val size = ((size shl 1) - 1).takeHighestOneBit() // next power of 2
     private val mask = this.size - 1
-    private val trace = arrayOfNulls<String>(this.size)
+    private val trace = arrayOfNulls<Any>(this.size)
     private val index = AtomicInteger(0)
 
-    override fun append(text: String) {
+    override fun append(arg1: Any) {
```

Comment:
All these methods are great, but they are internal and not available in user code, only `invoke {}` can be used from outside of atomicfu. Please fix that